### PR TITLE
feat(agents-api): Add `embed_instruction` argument to embeddings & document creation endpoints

### PR DESCRIPTION
**Pull request by HamadaSalhab**
> [!IMPORTANT]
> Adds `embed_instruction` argument to document creation and embedding endpoints, updating models and OpenAPI spec accordingly.
> 
>   - **Behavior**:
>     - Adds `embed_instruction` argument to `CreateDocRequest` and `EmbedQueryRequest` in `Docs.py`.
>     - Updates `create_doc()` in `create_doc.py` to handle `embed_instruction` by removing it from `doc_data`.
>     - Modifies `run_embed_docs_task()` in `create_doc.py` to pass `embed_instruction`.
>     - Updates `embed()` in `embed.py` to prepend `embed_instruction` to `text_to_embed`.
>   - **OpenAPI Specification**:
>     - Adds `embed_instruction` to `CreateDocRequest` and `EmbedQueryRequest` models in `models.tsp`.
>     - Updates `openapi-0.4.0.yaml` and `openapi-1.0.0.yaml` to include `embed_instruction` in required properties for `CreateDocRequest` and `EmbedQueryRequest`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=julep-ai%2Fjulep&utm_source=github&utm_medium=referral)<sup> for b68ab1d22ef2fc3062a66bb179c33d8099920944. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by creatorrr** [comment]
@HamadaSalhab You checked in the local jwt

**Comment by HamadaSalhab** [comment]
@creatorrr They were already there, the changes are there due to reformatting. Btw, we should delete these notebooks. I don't see the point in having them here.

**Review by Vedantsahai18 — APPROVED** [review]

**Review by whiterabbit1983 — APPROVED** [review]

File: agents-api/agents_api/routers/docs/embed.py [hunk 1]
```diff
@@ -22,6 +22,6 @@ async def embed(
         [text_to_embed] if isinstance(text_to_embed, str) else text_to_embed
     )
 
-    vectors = await litellm.aembedding(inputs=text_to_embed)
+    vectors = await litellm.aembedding(inputs=[data.embed_instruction] + text_to_embed)
```
Comment (creatorrr) [inline agents-api/agents_api/routers/docs/embed.py:25]: Why are you embedding the instructions separately? This looks like a but

Comment (HamadaSalhab) [reply agents-api/agents_api/routers/docs/embed.py]: You're right. Pushed a fix

**Pull request merged**